cpp/turbodbc_arrow/Library/src/arrow_result_set.cpp

```diff
@@ -224,6 +225,10 @@ Status arrow_result_set::fetch_all_native(std::shared_ptr<arrow::Table>* out)
                     break;
             }
         }
+        // Exit early if we are fetching one batch
+        if (batch_only) {
```

Comment:
Personally, I always have to think thrice what happens whenever I read a `break` statement outside of switch cases, and the comment above indicates that the code is probably not sufficiently clear itself. May I suggest the following:

* Extract what is in the do/while loop to a helper function, with the exception of the newly added `if` and the `fetch_next_batch()` call.
* Surround the loop with an if statement:
```
if (batch_only) {
    do {
        process_batch(...);
        rows_in_batch = base_result_.fetch_next_batch();
    } while (...);
} else {
    process_batch(...);
}
```

That would make the difference between batched and non-batched mode more obvious and make the overall function smaller by moving details to `process_batch(...)`.